sdks/python/apache_beam/runners/interactive/interactive_beam.py

```diff
@@ -314,6 +327,26 @@ def show(*pcolls, **configs):
   bcj.attempt_to_run_background_caching_job(
       runner, user_pipeline, user_pipeline.options)
 
+  pcolls = set(pcolls)
+  computed_pcolls = set()
+  for pcoll in pcolls:
+    if pcoll in ie.current_env().computed_pcollections:
```

Comment:
@KevinGG @rohdesamuel 
just for curiosity, how does a pcollection count as computed in streaming without source caching?

Reply:
All PCollections in the pipeline that is being executed will be marked as computed once a job (main job or background caching job) completes.

So it looks like this:
pcoll_a, pcoll_b belongs to pipeline
pipeline runs into job, returns a pipeline result (main job) or a BackgroundCachingJob (backgroun caching job).
Say pcoll_a and pcoll_b are not depending on each other, and the user invokes `show(pcoll_a)`.
A pipeline fragment that only contributes to pcoll_a will be executed.
Once pipeline result is in a successful state or BackgroundCachingJob.is_done(), pcoll_a is marked as computed.

Reply:
so for streaming pipelines that don't cache data, this will never finish. is that correct?